graphjet-core/src/main/java/com/twitter/graphjet/algorithms/counting/user/TopSecondDegreeByCountForUser.java

```diff
@@ -45,6 +46,13 @@ public TopSecondDegreeByCountForUser(
     StatsReceiver statsReceiver) {
     super(leftIndexedBipartiteGraph, expectedNodesToHit, statsReceiver);
   }
+  protected boolean isEdgeEngagementWithinAgeLimit(
+      TopSecondDegreeByCountRequestForUser request,
```

Comment:
It is hard to achieve this when the parent class abstracts the knowledge of TimestampeEdgeIterator. This method is shared between user and tweet recommendations, but for now the tweet side bipartite graph does not implement timestamps, so in the parent class, it cannot perform any operations related to timestamps. Therefore the only way to abstract timestamp information away in parent class is to pass "request" and "edgeIterator" as parameters, and cast them to class types that implement timestamps (in this case we cast EdgeIterator to TimestampEdgeIterator). This might sacrifice some encapsulation, but in the current design we must expose this much.